src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3692,40 +3687,23 @@ void mergePartialProperties(ref Dictionary<ReadOnlyMemory<char>, ImmutableArray<
                 }
                 else
                 {
-                    var (currentGet, prevGet) = ((SourcePropertyAccessorSymbol?)currentProperty.GetMethod, (SourcePropertyAccessorSymbol?)prevProperty.GetMethod);
-                    if (currentGet != null || prevGet != null)
-                    {
-                        var accessorName = (currentGet ?? prevGet)!.Name.AsMemory();
-                        mergeAccessors(ref membersByName, accessorName, currentGet, prevGet);
-                    }
-
-                    var (currentSet, prevSet) = ((SourcePropertyAccessorSymbol?)currentProperty.SetMethod, (SourcePropertyAccessorSymbol?)prevProperty.SetMethod);
-                    if (currentSet != null || prevSet != null)
-                    {
-                        var accessorName = (currentSet ?? prevSet)!.Name.AsMemory();
-                        mergeAccessors(ref membersByName, accessorName, currentSet, prevSet);
-                    }
-
-                    if ((object)membersByName == _lazyEarlyAttributeDecodingMembersDictionary)
-                    {
-                        // Avoid mutating the cached dictionary and especially avoid doing this possibly on multiple threads in parallel.
-                        membersByName = new Dictionary<ReadOnlyMemory<char>, ImmutableArray<Symbol>>(membersByName, ReadOnlyMemoryOfCharComparer.Instance);
-                    }
-
+                    DuplicateMembersByNameIfCached(ref membersByName);
```

Comment:
Previously, `membersByName` was potentially modified in `mergeAccessors` before duplicating.